src/dowel/csv_output.py

```diff
@@ -48,6 +49,36 @@ def record(self, data, prefix=''):
                            'Did you change key sets after your first '
                            'logger.log(TabularInput)?'.format(
                                set(self._fieldnames), set(to_csv.keys())))
+                # close existing log file
+                logFileName = self._log_file.name
+                self._log_file.close()
+                self._log_file = None
+                self._writer = None
+                # Read the existing data from the log file
+                f = open(logFileName)
+                reader = csv.DictReader(f)
+                fileFieldNames = reader.fieldnames
+                fileData = []
+                for row in reader:
+                    fileData.append(row)
+                reader = None
+                f.close()
+                # Clear the old file
+                f = open(logFileName, 'w+')
+                f.close()
+                # Create the new file and writer
+                self._log_file = open(logFileName, 'w') # copy from simple_outputs.py line 62
+                allFieldnames = set(self._fieldnames | to_csv.keys())
+                self._fieldnames = allFieldnames
+                self._writer = csv.DictWriter(
```

Comment:
It would probably be best to refactor out the writer initialization (and header writing) logic.

Reply:
What do you mean by "refactor out"? Do you mean to create another method that handles it and call that method?